tools/mksurfdata_esmf/src/mksurfdata.F90

```diff
@@ -1142,13 +1104,116 @@ subroutine normalize_and_check_landuse(ns_o)
             call shr_sys_abort()
          end if
 
-         ! Natural vegetated cover is 100% minus the sum of all other landunits
+         ! Determine the percent of each grid cell considered to be land. (See comments in
+         ! https://github.com/ESCOMP/CTSM/issues/1716 for details.)
+         !
+         ! Start by using the land fraction field from the PFT raw data set:
+         pct_land = pctlnd_pft(n)
+         !
```

Comment:
I appreciate you being very explicit in your thinking here. I think that is good, and will help when we look at this in the future. But, lots of text in comments is also problematic in code. I'm wondering if you could also have a TLDR section that lists the underlying assumptions in a really short way with bullet points?

Reply:
Good point. I think the tldr was really the first sentence ("But we don't want to..."). I have reorganized the comment a bit to highlight that. See the commit I just pushed.

Reply:
Perfect